docs/source/basic/terms.rst

```diff
@@ -0,0 +1,57 @@
+Terms
+=====
+
+This page provides an overview of the terms used in ``alpaka`` and the relationships between them.
+
+.. image:: /images/alpaka_terms.svg
+
+Platform
+--------
+
+- A ``platform`` contains information about the system, e.g. the available devices. 
+- Depending on the platform, it also contains a runtime context.
+- A ``platform`` has a least one device but it can also has many device.
+- Each ``platform`` can be used with N ``accelerator``. ``platforms`` and ``accelerator`` cannot freely combined. An ``accelerator`` supports only a specific ``platform``.
+ 
+Device
+------
+
+- A ``device`` represent a compute unit, such as a CPU or a GPU.
+- Each ``device`` is bounded to a specific ``platform``.
+- Each ``device`` can have N ``queues``.
+
+Accelerator
+-----------
+
+- A ``accelerator`` is a index mapping function. It distributes the index space to the chunks. The ``accelerator`` maps a continues index space to a blocked index domain decomposition.
+- It is not allowed to create an instance of an ``accelerator``.
+- A ``accelerator`` is bounded to a specific ``platform``.
+
+Queue
+-----
+
+- Stores operations which should be executed on a ``device``.
+- Operations can be ``TaskKernels``, ``Events``, ``Sets`` and ``Copies``.
```

Comment:
Also host tasks (that are not kernels).

Reply:
Memory allocations can also be submitted to a queue.

Reply:
Is host task also a trait/concept in alpaka? I never heared about this.

Reply:
Mhm, I'm not sure about a concept.

I think you can submit anything that can be called with no arguments:
```c++
#include <iostream>

#include <alpaka/alpaka.hpp>

int main(void) {

#ifdef ALPAKA_ACC_GPU_CUDA_ENABLED
  using Platform = alpaka::PlatformCudaRt;
#else
  using Platform = alpaka::PlatformCpu;
#endif  // ALPAKA_ACC_GPU_CUDA_ENABLED
  Platform platform;

  using Device = alpaka::Dev<Platform>;
  Device device = alpaka::getDevByIdx(platform, 0u);

  using Queue = alpaka::Queue<Device, alpaka::NonBlocking>;
  Queue queue{device};

  std::cout << "before submitting\n";

  alpaka::enqueue(queue, [](){
    // asynchronous host task executed from a device queue
    std::cout << "async hello world from " << alpaka::core::demangled<Device> << '\n';
  });

  std::cout << "after submitting, before waiting\n";

  alpaka::wait(queue);

  std::cout << "after waiting\n";

  return 0;
}
```

Reply:
I was not aware, that is possible. I use a task kernel and a cpu device for it. But I found the feature in the queue tests.

I will add it. Looks like my documentation makes sense ;-)

Reply:
> I use a task kernel and a cpu device for it.

That also works. 

The point of using a host task with a device queue is to run the host code only after the device operations are complete.

A similar functionality in CUDA is provided by `cudaLaunchHostFunc`, but the alpaka version does not have the limitation that it cannot call other device API functions.